src/test-minconf.cpp

```diff
@@ -2,6 +2,8 @@
 #ifndef __sun // exclude solaris (https://github.com/r-lib/testthat/issues/1257)
 #include "test-minconf.h"
 
+const int TestMinConf::NA; //Definition
```

Comment:
This doesn't make much sense here ;-) NA is already defined here: https://github.com/r-spatialecology/spectre/blob/6f4484e61e1a4a2ebfe2f70140357385635d39bc/src/minconf.h#L21

The comment is not about that we forgot to define NA, but that the abs()-value here https://github.com/r-spatialecology/spectre/blob/6f4484e61e1a4a2ebfe2f70140357385635d39bc/src/minconf.cpp#L326 could become NA (which could be a negative value) and added to an unsigned integer. However, this should theoretically not happen because of https://github.com/r-spatialecology/spectre/blob/6f4484e61e1a4a2ebfe2f70140357385635d39bc/src/minconf.cpp#L322

Reply:
Do we need to use `NA_REAL` and `is_na()`? Does that might help? I didn't really look at the code and I am only a smartass here sorry 😄

Reply:
If I'm not mistaken, the error occurs when `std::abs(commonness[site][other_site] - target[site][other_site])` is equal to `std::abs(-2147483648)`, because the maximum positive value an integer can hold is 214748364**7**. The default NA we use is indeed -2147483648, so the issue would occur if commonness[site][other_site] is NA and target[site][other_site] is 0 at the same time. This should not happen, however, because target and commonnes should have NA at exact the same places.

But as our tests did not provoke the error, I don't know if my reasoning is correct and we need to reproduce this.

Reply:
Ok. So is the best bet to contact CRAN (Uwe) and try ask them how to reproduce this error?

Reply:
Yes, that would help a lot